spec/features/bookmarks_spec.rb

```diff
@@ -96,4 +107,26 @@
       find('.blacklight-icons-bookmark').click
     end
   end
+
+  context "without bookmark navbar element" do
+    let!(:default_nav_partials) { CatalogController.blacklight_config.navbar.partials }
+
+    before do
+      CatalogController.blacklight_config.navbar.partials = CatalogController.blacklight_config.navbar.partials.except(:bookmark)
+    end
+
+    after do
+      CatalogController.blacklight_config.navbar.partials = default_nav_partials
+    end
+
+    it 'adds and removes bookmarks', :js do
+      visit solr_document_path('2007020969')
+      expect(page).to have_no_css('#bookmarks_nav')
+      check 'Bookmark'
+
```

Comment:
Can you add a check here before navigating away from the page that shows it has been saved? Typically you can look for `In Bookmarks` in the page content.

Reply:
Thanks, good call.

 I have added that check, but it fails the rails 6 check. I built an internal test app using rails 6.1.7.8 and encountered the same failure. When inspecting the app, it looks like the  `BookmarkToggle` event handler callback is fired twice in succession, so the checkbox checks and unchecks itself with one click. I haven't pinpointed the issue, but perhaps it's related to how the js assets are built/handled when using rails 6?

Reply:
Some additional context: 

In the rails `6.1.7.8` test app, the browser receives both a `blacklight.source.js` and `blacklight.debug.js` file that appear the same. The `clicked` function is called in both these js files according to the debugger. This has the effect of the checkbox being clicked twice in succession, which reflects the failing test.  While I am unfamiliar with the asset pipeline for rails 6 and BL, I will investigate [turning off debug mode](https://guides.rubyonrails.org/v4.0.0/asset_pipeline.html#turning-debugging-off).